src/edge/recombine.py

```diff
@@ -510,6 +510,11 @@ def find_root_genome(genome):
 
 
 def lock_genome(genome):
+    """
+    Do a select for update, which, when executed inside a transaction, places a
```

Comment:
FYI I only spent two minutes poking into this, but from what I could tell, the places where you call `lock_genome` aren't inside transactions. Unless they're happening higher up in the call chain...

Reply:
Correct. It's upstream caller that's calling the atomic, because you need the locking and the rest of the work to happen all within a transaction. Should I change the method name to lock_genome_call_if_in_transaction? or you think the comment is sufficient.